### DISCUSSION TITLE
ScriptReference/MaterialPropertyBlock

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#62] [Notes]
# ScriptReference/MaterialPropertyBlock

https://docs.unity3d.com/ScriptReference/MaterialPropertyBlock.html

**Comment by vertxxyz** [comment]
If you're looking to set Tiling or Offset properties use the "_ST" suffix
```csharp
propertyBlock.SetVector("_MainTex_ST", new Vector4(0.5f, 0.5f, 0.25f, 0.25f));
```
See [Special Texture properties](https://docs.unity3d.com/Manual/SL-PropertiesInPrograms.html) - Texture tiling & offset
